src/vm/vm_core.rs

```diff
@@ -483,8 +515,13 @@ impl VirtualMachine {
         exec_scopes: &mut ExecutionScopes,
         hint_data_dictionary: &HashMap<usize, Vec<Box<dyn Any>>>,
         constants: &HashMap<String, BigInt>,
+        hooks: &Vec<Hook>,
     ) -> Result<(), VirtualMachineError> {
         self.step_hint(hint_executor, exec_scopes, hint_data_dictionary, constants)?;
+        let instruction = self.decode_current_instruction()?;
+        for hook in hooks {
+            hook.execute(self, &instruction, exec_scopes, constants)?
+        }
         self.step_instruction()
```

Comment:
Execute only if hook was no called

Reply:
skep_next_instruction work the oposite way